platforms/allwinner-d1/README.md

```diff
@@ -105,6 +111,78 @@ building `xfel` from source for Linux, MacOS, and Windows can be found
 [`xfel-nix`]: https://github.com/hawkw/dotfiles/blob/736d80487687b0610a1b17f5bbec6b22a501207c/nixos/pkgs/xfel.nix
 [`xfel-nix-udev`]: https://github.com/hawkw/dotfiles/blob/736d80487687b0610a1b17f5bbec6b22a501207c/nixos/machines/noctis.nix#L102-L104
 
+## Boot Procedure
+On reset, the D1 executes its internal *Boot ROM* (`BROM`), which either loads
+a first stage bootloader or enters *FEL mode*.
+
+The `BROM` will:
+* Do some initial setup of the clocks
+* Check the *FEL pin*: if it is low (connected to GND), it will enter FEL mode
+* Check the SD card (SMHC0) and connected SPI flash for a valid [eGON header]
+* Fall back to FEL mode if no valid header is found
+
+You can find additional info about [`BROM`] and [FEL] on the *linux-sunxi* wiki.
+In short, in FEL mode the D1 will present itself as a USB device and allow
+(using a custom protocol) things like reading data, writing data and starting
+code execution.
+Different tools like [`sunxi-fel`] and [`xfel`] have been developed that speak
+this protocol and implement functionality like initializing DRAM
+(by loading code that does this into SRAM and then executing it).
+However, this is all volatile, so these actions have to be repeated on reset.
+
+To have a persistent boot, we can use one of the media that is probed
+by the `BROM` for an eGON header: the SD card or SPI flash
+(if you have this on your board).
+On this persistent medium, a first stage bootloader needs to be present,
+preceded by a valid eGON header.
+The `BROM` will then load it into SRAM and execute it.
+
+This bootloader has to, at a minimum, initialize DRAM and load either
+a second stage bootloader or the actual application (in our case MnemOS)
+into DRAM so it can transfer control of execution to it.
+
+### DRAM initialization
+The initialization code for the DDR3 RAM is somewhat of a
+[black box][sunxi wiki DRAM]. It is hard to determine who was first in
+reverse-engineering the necessary Allwinner blobs, but one candidate
+is [this][pnru boot0], which served as a basis for the original
+[SPL work][sun20i_d1_spl] by smaeul. This is now deprecated, as it is included
+in smaeul's [u-boot fork][u-boot mctl], which will hopefully be merged upstream
+one day.
+
+[Oreboot] has a [Rust port][oreboot mctl] of this code.
+TODO: figure out why some of the DRAM parameters depend on
+which board is selected.
+
+### SD card layout
+The *linux-sunxi* wiki has some [more information][sdcard-layout] on
+the required layout of the SD card.
+The eGON header with first stage bootloader has to be located at an offset
+of either 8KB or 128KB, in order to leave some space for a partition table
+(so you can have, e.g., a FAT filesystem on your SD card at the same time).
+
+To prepare your SD card, you could do the following:
+```sh
+sudo dd if=first-stage-boot.bin of=/dev/sdX bs=1024 seek=8
+sudo dd if=mnemos.bin of=/dev/sdX bs=1024 seek=40
```

Comment:
Writing the bootloader to the SD card only needs to happen once, so that could be a manual step for now,
but automatically writing mnemos to an SD card would be nice!